sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -136,8 +136,8 @@ public async Task StopAsync(CancellationToken cancellationToken)
                             await _messageProcessor.Value.Processor.StopProcessingAsync(cancellationToken).ConfigureAwait(false);
                         }
                     }
-                    // Batch processing will be stopped via the _started flag on its next iteration
-
+                    // Batch processing will be stopped via the _cancellationTokenSource on its next iteration
+                    _cancellationTokenSource.Cancel();
                     _started = false;
```

Comment:
do we still need the flag?

Reply:
It is used in other places, so I didn't want to mess with that.